scalableminds/webknossos
# Optional usage of existing segmentation layer in hybrid tracings

This PR makes it optional to use the existing segmentation layer for hybrid tracings. The behavior should be the same compared to volume tracings.

Please note the following trade-off I decided on: When viewing a dataset in view mode, it is possible to create a tracing from there on, but only if you are logged in. To not edit the login model that is displayed when the user clicks `Create Tracing` without being logged in, the tracing will only open up when selecting the button again after logging in. Keeping the dropdown open while the user uses the login model to log in, did not work out :confused: 

### URL of deployed dev instance (used for testing):
- [https://optionalreusageofsegmentationforhybrid.webknossos.xyz](https://optionalreusageofsegmentationforhybrid.webknossos.xyz)

### Steps to test:
- Logout, open a tracing in view mode and use the `Create  Tracing` Button.
- The login modal should open and ask you for logging in. Log in and use the `Create Tracing` button again. Now you should be able to select between using the existing segmentation layer or creating a new one. 
- Stay logged in and reopen any dataset in view mode. Now use the `Create Tracing` button again. The dropdown should now show up immediately and the creation of a tracing should work as usual.

- Open the dataset listing view in the dashboard (`Datasets`). From there open a hybrid tracing (explorational). The dropdown should be the same as for volume tracings and the creation should work as expected.

### Issues:
- fixes #4022

------
